### PR TITLE
SankeyChart: Refactoring for improved minification

### PR DESCRIPTION
**Pull request by StarTether**
## New Behavior

I expect the file size to shrink a little bit because we are accessing `this.` in fewer locations. `private` methods which do not access `this` have been moved outside of the class entirely.

Some comments have been removed from the file because they are no longer necessary to suppress `lint` issues. Other comments have been added to flag what look like embedded user-readable text and emphasize it for localizing.

## Related Issue(s)

- Fixes #

### CODE REVIEW

File: packages/web-components/docs/api-report.md [hunk 1]
```diff
@@ -1,1734 +1,1734 @@
...
-// dist/dts/custom-elements.d.ts:57:5 - (ae-incompatible-release-tags) The symbol "fluentButton" is marked as @public, but its signature references "Button" which is marked as @internal
-// dist/dts/custom-elements.d.ts:96:5 - (ae-incompatible-release-tags) The symbol "fluentTextArea" is marked as @public, but its signature references "TextArea" which is marked as @internal
-// dist/dts/custom-elements.d.ts:97:5 - (ae-incompatible-release-tags) The symbol "fluentTextField" is marked as @public, but its signature references "TextField" which is marked as @internal
-// dist/dts/custom-elements.d.ts:98:5 - (ae-incompatible-release-tags) The symbol "fluentToolbar" is marked as @public, but its signature references "Toolbar" which is marked as @internal
-// dist/dts/custom-elements.d.ts:99:5 - (ae-incompatible-release-tags) The symbol "fluentTooltip" is marked as @public, but its signature references "Tooltip" which is marked as @internal
-
-// (No @packageDocumentation comment for this package)
-
-```
+## API Report File for "@fluentui/web-components"
```
Comment (AtishayMsft) [inline packages/web-components/docs/api-report.md:1]: This change is unrelated to Sankey chart. Should be reverted

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/2314a84349b2e58b54ebe70d4408ef5fc060f43f'>2314a84349b2e58b54ebe70d4408ef5fc060f43f</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-import { ChartHoverCard, IBasestate, IChartHoverCardProps, SLink, SNode } from '../../index';
-import { Callout, DirectionalHint } from '@fluentui/react/lib/Callout';
+import { SankeyGraph, SankeyLayout, sankey as d3Sankey, sankeyJustify, sankeyRight } from 'd3-sankey';
 import { select, selectAll } from 'd3-selection';
-import { FocusZone, FocusZoneDirection, FocusZoneTabbableElements } from '@fluentui/react-focus';
+import { area as d3Area, curveBumpX as d3CurveBasis } from 'd3-shape';
+import * as React from 'react';
+import { ChartHoverCard, IBasestate, IChartHoverCardProps, SLink, SNode } from '../../index';
 import { IMargins } from '../../utilities/utilities';
+import { ISankeyChartData, ISankeyChartProps, ISankeyChartStyleProps, ISankeyChartStyles } from './SankeyChart.types';
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx:12]: We have another open task to reduce the overall bundle size for the charting package.
[https://github.com/microsoft/fluentui/pull/30244](https://github.com/microsoft/fluentui/pull/30244)

We are starting with obfuscating the tree chart and plan to do it for Sankey chart next.

**Review by AtishayMsft — APPROVED** [review]
:shipit:

File: packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx (hunk 2 above)
Comment (StarTether) [reply packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx:12]: That sounds good. I am not trying to do absolute bundle size mitigation, here. I am just tackling some low-hanging fruit that I would normally fix anyway. I am doing this as a separate PR so that these changes do not cloud my next PR.

File: packages/web-components/docs/api-report.md (hunk 1 above)
Comment (StarTether) [reply packages/web-components/docs/api-report.md:1]: I would like to, but this is a LF to CRLF change introduced via a `git add *.* --renormalize`

Comment (AtishayMsft) [reply packages/web-components/docs/api-report.md:1]: This is an autogenerated file and should not be edited manually. Could you revert it.
image
